hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractDeleteHandler.java

```diff
@@ -64,6 +65,12 @@ public ContractDeleteHandler() {
         // Exists for injection
     }
 
+    @Override
+    public void pureChecks(@NonNull TransactionBody txn) throws PreCheckException {
+        final var op = txn.contractDeleteInstanceOrThrow();
+        mustExist(op.contractID(), INVALID_CONTRACT_ID);
```

Comment:
I note that at `TokenDeleteHandler.pureChecks` it does this check by calling `PreCheckException.validateTruePreCheck` and at `CryptoDeleteHandler` it directly throws a `PreCheckException` (as do several other `pureChecks` - someday this could be cleaned up to have _one_ way to do it (which would be more likely to be correct or at least easier to use as a template for future work).

Reply:
I agree. We need to go through all handlers and standardize such patterns.